codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-query-async-client-class.java

```diff
@@ -0,0 +1,271 @@
+            return CompletableFutureUtils.failedFuture(t);
+        }
+    }
+
+    /**
+     * Some operation with a streaming output
+     *
+     * @param streamingOutputOperationRequest
+     * @param asyncResponseTransformer
+     *        The response transformer for processing the streaming response in a non-blocking manner. See
+     *        {@link AsyncResponseTransformer} for details on how this callback should be implemented and for links to
+     *        precanned implementations for common scenarios like downloading to a file. The service documentation for
+     *        the response content is as follows 'This be a stream'.
+     * @return A future to the transformed result of the AsyncResponseTransformer.<br/>
+     *         The CompletableFuture returned by this method can be completed exceptionally with the following
+     *         exceptions.
+     *         <ul>
+     *         <li>SdkException Base class for all exceptions that can be thrown by the SDK (both service and client).
+     *         Can be used for catch all scenarios.</li>
+     *         <li>SdkClientException If any client side error occurs such as an IO related failure, failure to get
+     *         credentials, etc.</li>
+     *         <li>QueryException Base class for all service exceptions. Unknown exceptions will be thrown as an
+     *         instance of this type.</li>
+     *         </ul>
+     * @sample QueryAsyncClient.StreamingOutputOperation
+     * @see <a href="http://docs.aws.amazon.com/goto/WebAPI/query-service-2010-05-08/StreamingOutputOperation"
+     *      target="_top">AWS API Documentation</a>
+     */
+    @Override
+    public <ReturnT> CompletableFuture<ReturnT> streamingOutputOperation(
+        StreamingOutputOperationRequest streamingOutputOperationRequest,
+        AsyncResponseTransformer<StreamingOutputOperationResponse, ReturnT> asyncResponseTransformer) {
+        try {
+
+            HttpResponseHandler<StreamingOutputOperationResponse> responseHandler = protocolFactory
+                .createResponseHandler(StreamingOutputOperationResponse::builder);
+
+            HttpResponseHandler<AwsServiceException> errorResponseHandler = protocolFactory.createErrorResponseHandler();
+
+            CompletableFuture<ReturnT> executeFuture = clientHandler.execute(
+                new ClientExecutionParams<StreamingOutputOperationRequest, StreamingOutputOperationResponse>()
+                    .withOperationName("StreamingOutputOperation")
+                    .withMarshaller(new StreamingOutputOperationRequestMarshaller(protocolFactory))
+                    .withResponseHandler(responseHandler).withErrorResponseHandler(errorResponseHandler)
+                    .withInput(streamingOutputOperationRequest), asyncResponseTransformer);
+            executeFuture.whenComplete((r, e) -> {
+                if (e != null) {
+                    asyncResponseTransformer.exceptionOccurred(e);
+                }
+            });
```

Comment:
should we return the result of executeFuture.whenComplete? I think the only difference is that if exceptionOccurred throws an exception, the exception is propagated to the result future.

Reply:
Maybe we should do `runAndLogError` like line 252. Customers probably don't care if exception occurs in `exceptionOccurred`.

Reply:
Sure! I don't think exceptionOccurred is supposed to raise an exception anyway, right?

Reply:
Right, at least not in our built-in `AsyncResponseTransformer`s. I'll update the javadoc for `exceptionOccurred` to make it more clearly.